### PR TITLE
Add a new 'configure-pull-request' composite GitHub action

### PR DESCRIPTION
**Pull request by clementnero**
Related to https://github.com/camunda/team-infrastructure/issues/204

Add a new composite GitHub action to configure a Pull Request. It is part of the refactoring of the `maintenance_project.yml` workflow but the logic can be reused for other needs.

### CODE REVIEW

File: configure-pull-request/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+jobs:
+  configure-pr:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: camunda/infra-global-github-actions/configure-pull-request@main
+      with:
+        github-token: ${{ secrets.MY_PAT }}
+        labels: label-1,label-2
+        project-url: https://github.com/orgs/camunda/projects/project-id/
+        reviewsers: user1,user2
```
Comment (maxdanilov) [inline configure-pull-request/README.md:38]:
```suggestion
        reviewers: user1,user2
```

File: configure-pull-request/action.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: Configure Pull Request
+
+description: Sets the parameters of an existing Pull Request at once
+
+inputs:
+  github-token:
+    description: A GitHub personal access token with write access to the project
```
Comment (maxdanilov) [inline configure-pull-request/action.yml:7]:
```suggestion
    description: A GitHub access token with write access to the project
```
A PAT is something bound to a user account; access tokens would include both PATs and dynamic tokens like the ones obtained with [tibdex/github-app-token](https://github.com/camunda/github-actions-recipes/blob/main/README.md#authenticate-from-a-github-app-via-tibdexgithub-app-token-usages-in-camunda)

File: configure-pull-request/action.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+runs:
+  using: composite
+  steps:
+    - name: Set labels
+      uses: buildsville/add-remove-label@eeae411a9be2e173f2420e1644514edbecc4e835 # pin@v2.0.0
+      with:
+        labels: ${{ inputs.labels }}
+        token: ${{ inputs.github-token }}
+        type: add
+      if: ${{ inputs.labels != '' }}
```
Comment (maxdanilov) [inline configure-pull-request/action.yml:31]: I would suggest moving the `if` condition right under `name` so it's immediately clear that this step is conditional

File: configure-pull-request/action.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Set project
+      uses: actions/add-to-project@960fbad431afda394cfcf8743445e741acd19e85 # pin@v0.4.0
+      with:
+        github-token: ${{ inputs.github-token }}
+        project-url: ${{ inputs.project-url }}
+      if: ${{ inputs.project-url != '' }}
```
Comment (maxdanilov) [inline configure-pull-request/action.yml:37]: I would suggest moving the `if` condition right under `name` so it's immediately clear that this step is conditional

File: configure-pull-request/action.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Set reviewers (users)
+      uses: dannysauer/actions-assigner@ca2ca780928294ee5126986cc31db6c2ee7d69b0 # pin@v2.0.1
+      with:
+        reviewers: ${{ inputs.reviewers }}
+        token: ${{ inputs.github-token }}
+      if: ${{ inputs.reviewers != '' }}
```
Comment (maxdanilov) [inline configure-pull-request/action.yml:43]: I would suggest moving the `if` condition right under `name` so it's immediately clear that this step is conditional

File: configure-pull-request/action.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Set reviewers (teams)
+      uses: dannysauer/actions-assigner@ca2ca780928294ee5126986cc31db6c2ee7d69b0 # pin@v2.0.1
+      with:
+        team-reviewers: ${{ inputs.team-reviewers }}
+        token: ${{ inputs.github-token }}
+      if: ${{ inputs.team-reviewers != '' }}
```
Comment (maxdanilov) [inline configure-pull-request/action.yml:49]: I would suggest moving the `if` condition right under `name` so it's immediately clear that this step is conditional

File: configure-pull-request/action.yml (hunk 2 above)
Comment (maxdanilov) [inline configure-pull-request/action.yml:7]: I'd suggest adding explicit types for the inputs (e.g., `type: string`): https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-of-onworkflow_callinputs

Currently, GH ignores that in runtime, but this behavior may be useful in the future and provides a self-documenting code.

File: configure-pull-request/action.yml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        project-url: ${{ inputs.project-url }}
+      if: ${{ inputs.project-url != '' }}
+    - name: Set reviewers (users)
+      uses: dannysauer/actions-assigner@ca2ca780928294ee5126986cc31db6c2ee7d69b0 # pin@v2.0.1
+      with:
+        reviewers: ${{ inputs.reviewers }}
+        token: ${{ inputs.github-token }}
+      if: ${{ inputs.reviewers != '' }}
+    - name: Set reviewers (teams)
+      uses: dannysauer/actions-assigner@ca2ca780928294ee5126986cc31db6c2ee7d69b0 # pin@v2.0.1
```
Comment (maxdanilov) [inline configure-pull-request/action.yml:45]: will Renovate update this action in the future if a new version is released or will it be pinned to 2.0.1 permanently?

**Review by maxdanilov — CHANGES_REQUESTED** [review]

File: configure-pull-request/action.yml (hunk 2 above)
Comment (clementnero) [reply configure-pull-request/action.yml]: Input type seems not to be allowed for composite action. I will try.

File: configure-pull-request/action.yml (hunk 7 above)
Comment (clementnero) [reply configure-pull-request/action.yml:46]: I am not familiar with Renovate configuration but it seems that digest can be updated. What is the Infra team strategy to deal with (small) external action ?

Comment (maxdanilov) [reply configure-pull-request/action.yml:51]: We usually just specify the semver (i.e., `uses: dannysauer/actions-assigner@2.0.1`)

Renovate also supports hash pinning with following the updates: https://docs.renovatebot.com/modules/manager/github-actions/#additional-information

File: configure-pull-request/action.yml (hunk 2 above)
Comment (maxdanilov) [reply configure-pull-request/action.yml]: ah I didn't know. Then let's not do it.

Comment (clementnero) [reply configure-pull-request/action.yml]: I have added the types for the inputs as suggested (it works, you were rights)

File: configure-pull-request/action.yml (hunk 7 above)
Comment (clementnero) [reply configure-pull-request/action.yml:51]: > We usually just specify the semver (i.e., `uses: dannysauer/actions-assigner@2.0.1`) and Renovate creates PR when a new version is released.
> 
> Renovate also supports hash pinning with following the updates: https://docs.renovatebot.com/modules/manager/github-actions/#additional-information

I will specify the semver for now. Thx for the link.

**Review by maxdanilov — APPROVED** [review]
Looks great, pretty lean!

**Pull request merged**